src/libmain/progress-bar.cc

```diff
@@ -412,7 +437,7 @@ class ProgressBar : public Logger
         auto width = getWindowSize().second;
         if (width <= 0) width = std::numeric_limits<decltype(width)>::max();
 
-        writeToStderr("\r" + filterANSIEscapes(line, false, width) + ANSI_NORMAL + "\e[K");
+        output = "\r" + filterANSIEscapes(line, false, width) + ANSI_NORMAL + "\e[K";
```

Comment:
Do we actually need the new `draw()` function? Can't we just do
```suggestion
        auto output = "\r" + filterANSIEscapes(line, false, width) + ANSI_NORMAL + "\e[K";
        if (output != lastOutput) {
            writeToStderr(*newOutput);
            lastOutput = std::move(*newOutput);
       }
```

Reply:
Done. I've inlined the new `draw` back into the old `draw`, so that's a single function again.
I've also got rid of the `output` local, as we don't need a mutable variable for that anyway.